core/deployment/src/main/java/io/quarkus/deployment/builditem/GeneratedClassBuildItem.java

```diff
@@ -18,7 +18,7 @@ public GeneratedClassBuildItem(boolean applicationClass, String name, byte[] cla
             throw new IllegalArgumentException("Name cannot start with '/':" + name);
         }
         this.applicationClass = applicationClass;
-        this.name = name;
+        this.name = name.replace('/', '.');
```

Comment:
Hmmm, we aren't using the `getName()` anywhere? It seems a bit odd to do that here?

Reply:
This could potentially be breaking, so we probably only want to do it in `toString` for now

Reply:
It is not breaking. If you catch this list in a debugger you will see that about half use `/` and half use `.`. But using `.` is correct. Using`/` works in some situations though, including this one.

Reply:
My point is that the call sites might depend on how this exist now. I believe the safest course of action would be to add a property called `sanitizedName` or something like that

Reply:
I see plenty of usages of `getName()` in the codebase, which some do `getName().replace('/', '.')` and some do `getName().replace('.', '/')`  (depending on what the specific codepath is trying to do).

It seems to me that the safest option would be to deprecate `name` and introduce `className` and `binaryName` (or whatever we think are the best names) and update the various call sites. 
I would also leave `name` for now as there are certainly extensions not in ther core repo that use it.

Reply:
OK I'll come up with something.

Reply:
🙏🏽